common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -18,269 +20,302 @@
-            }
-            if (envVariablePresent("JOB_NAME")) {
-                addCustomValueAndSearchLink(buildScan, "CI job", envVariable("JOB_NAME"));
-            }
-            if (envVariablePresent("STAGE_NAME")) {
-                addCustomValueAndSearchLink(buildScan, "CI stage", envVariable("STAGE_NAME"));
-            }
+        private String projectProperty(String name) {
+            return (String) gradle.getRootProject().findProperty(name);
         }
 
-        if (isTeamCity()) {
-            if (envVariablePresent("BUILD_URL")) {
-                buildScan.link("TeamCity build", envVariable("BUILD_URL"));
-            }
-            if (envVariablePresent("BUILD_NUMBER")) {
-                buildScan.value("CI build number", envVariable("BUILD_NUMBER"));
-            }
-            if (envVariablePresent("BUILD_AGENT_NAME")) {
-                addCustomValueAndSearchLink(buildScan, "CI agent", envVariable("BUILD_AGENT_NAME"));
-            }
+        public void enhance() {
+            captureOs();
+            captureIde();
+            captureCiOrLocal();
+            captureCiMetadata();
+            captureGitMetadata();
+            captureTestParallelization();
         }
 
-        if (isCircleCI()) {
-            if (envVariablePresent("CIRCLE_BUILD_URL")) {
-                buildScan.link("CircleCI build", envVariable("CIRCLE_BUILD_URL"));
-            }
-            if (envVariablePresent("CIRCLE_BUILD_NUM")) {
-                buildScan.value("CI build number", envVariable("CIRCLE_BUILD_NUM"));
-            }
-            if (envVariablePresent("CIRCLE_JOB")) {
-                addCustomValueAndSearchLink(buildScan, "CI job", envVariable("CIRCLE_JOB"));
-            }
-            if (envVariablePresent("CIRCLE_WORKFLOW_ID")) {
-                addCustomValueAndSearchLink(buildScan, "CI workflow", envVariable("CIRCLE_WORKFLOW_ID"));
-            }
+        private void captureOs() {
+            buildScan.tag(sysProperty("os.name"));
         }
 
-        if (isBamboo()) {
-            if (envVariablePresent("bamboo_resultsUrl")) {
-                buildScan.link("Bamboo build", envVariable("bamboo_resultsUrl"));
-            }
-            if (envVariablePresent("bamboo_buildNumber")) {
-                buildScan.value("CI build number", envVariable("bamboo_buildNumber"));
-            }
-            if (envVariablePresent("bamboo_planName")) {
-                addCustomValueAndSearchLink(buildScan, "CI plan", envVariable("bamboo_planName"));
-            }
-            if (envVariablePresent("bamboo_buildPlanName")) {
-                addCustomValueAndSearchLink(buildScan, "CI build plan", envVariable("bamboo_buildPlanName"));
-            }
-            if (envVariablePresent("bamboo_agentId")) {
-                addCustomValueAndSearchLink(buildScan, "CI agent", envVariable("bamboo_agentId"));
-            }
+        private void captureIde() {
+            gradle.projectsEvaluated(g -> {
```

Comment:
oh, is this because project properties are not yet available when this is applied as a settings plugin?
maybe we should add a comment here so we know why it is there.